### ISSUE TITLE
Isolated sewer vault

### ISSUE THREAD

**Issue by nojaa3** [crawl/crawl#1943]
"webtiles" using safari 13.1.2

DCSS VERSION AND GAME INFORMATION:
0.27-a0-1139-g28e009f8a6
Sever vault generated isolated from rest of the dungeon, requiring digging wand to access. Pictures are after taking magical mapping.

![Näyttökuva 2021-5-7 kello 17 13 11](https://user-images.githubusercontent.com/82903954/117464100-04a7fb80-af59-11eb-92e4-80d6a8887a3e.png)
![Näyttökuva 2021-5-7 kello 17 24 53](https://user-images.githubusercontent.com/82903954/117464357-3de06b80-af59-11eb-9a96-aad2e3c6fa49.png)

**Comment by DreamDustDCSS** [comment]
Pretty sure this is the vault:
```
NAME:   portal_sewer_entry_b
ORIENT: float
: sewer_portal(_G)
SUBST:  ' = W. , W = WW1
KMONS:  1 = rat w:20 / river rat / ball python / adder w:2 / \
            giant cockroach / worm w:3
MAP
xxxxxx
xOWWW'
xWWWW.
xWWW'.
xWWW..
xW'...
ENDMAP
```
Very curious since although it doesn't have an explicit entry point (like a @ or + glyph), I thought the level generator is supposed to use other traversable tiles (like the floor tiles here) on the edge of a vault as an entry point to connect it to the rest of the level. 

I'm just a random person with no real knowledge of how the level generator works, but I'm guessing there's something wrong with the implicit entry point logic. Either that or the sewer entry portal is tricking the level generator into thinking this vault is still accessible despite being walled off - as if the portal is a stair or something.

A quick fix for this particular vault might be to add an explicit entry point. Like this:
```
MAP
xxxxxx
xOWWW'
xWWWW.
xWWW'.
xWWW..
xW'..@
ENDMAP
```
But again, this wouldn't fix the deeper issue (which is probably affecting a bunch of other vaults as well).

**Comment by rawlins** [comment]
This issue should be fixed by: 940ed4fa3142e

Having explicit `@` glyphs (and/or `transparent`) is very often better, but in this case it should now be interchangable to leave it up to levelgen.
